python/ray/tune/integration/lightgbm.py

```diff
@@ -1,211 +1,16 @@
-            eval_result[data_name][eval_name + suffix] = result
-            if stdv is not None:
-                eval_result[data_name][eval_name + "-stdv"] = stdv
-        return eval_result
-
-    @staticmethod
-    def _create_checkpoint(model: Booster, epoch: int, filename: str, frequency: int):
-        # Deprecate: Raise error in Ray 2.8
-        if log_once("lightgbm_ray_legacy"):
-            warnings.warn(
-                "You are using an outdated version of LightGBM-Ray that won't be "
-                "compatible with future releases of Ray. Please update LightGBM-Ray "
-                "with `pip install -U lightgbm_ray`."
-            )
-
-        if not frequency or epoch % frequency > 0 or (not epoch and frequency > 1):
-            # Skip 0th checkpoint if frequency > 1
-            return
-        with tune.checkpoint_dir(step=epoch) as checkpoint_dir:
-            model.save_model(os.path.join(checkpoint_dir, filename))
-
-    @contextmanager
-    def _get_checkpoint(
-        self, model: Booster, epoch: int, filename: str, frequency: int
-    ) -> Optional[Checkpoint]:
-        if not frequency or epoch % frequency > 0 or (not epoch and frequency > 1):
-            # Skip 0th checkpoint if frequency > 1
-            yield None
-            return
-
-        with tempfile.TemporaryDirectory() as checkpoint_dir:
-            model.save_model(os.path.join(checkpoint_dir, filename))
-            checkpoint = Checkpoint.from_directory(checkpoint_dir)
-            yield checkpoint
-
-    def __call__(self, env: CallbackEnv) -> None:
-        if self._frequency > 0 and self._checkpoint_callback_cls:
-            self._checkpoint_callback_cls.__call__(self, env)
-        if self._report_callback_cls:
-            # Deprecate: Raise error in Ray 2.8
-            if log_once("xgboost_ray_legacy"):
-                warnings.warn(
-                    "You are using an outdated version of LightGBM-Ray that won't be "
-                    "compatible with future releases of Ray. Please update LightGBM-Ray"
-                    " with `pip install -U lightgbm_ray`."
-                )
-
-            self._report_callback_cls.__call__(self, env)
-            return
-
-        with self._get_checkpoint(
-            model=env.model,
-            epoch=env.iteration,
-            filename=self._filename,
-            frequency=self._frequency,
-        ) as checkpoint:
-            eval_result = self._get_eval_result(env)
-            report_dict = self._get_report_dict(eval_result)
-            train.report(report_dict, checkpoint=checkpoint)
-
-
-class _TuneCheckpointCallback(TuneCallback):
-    def __init__(self, *args, **kwargs):
-        raise DeprecationWarning(
-            "`ray.tune.integration.lightgbm._TuneCheckpointCallback` is deprecated."
-        )
+# If a user imports from this package, then the class module should line up.
+TuneReportCheckpointCallback.__module__ = "ray.tune.integration.lightgbm"
```

Comment:
nit: If we're aliasing in this direction I don't know if we need to do this. I think it was primarily for aliasing in the "new" direction. Though it should be used infrequently enough that this doesn't make a big difference either way (I don't think it'll break anything).